[CR] Fixed "Snow glare is active during night time #33413"

ipcyborg opened the pull request:
#### Summary
```SUMMARY: Bugfixes "Snow glare is active during night time #33413"```

#### Purpose of change
Fixes #33413

#### Describe the solution
I can't find any way to detect if there is snow on the tile.
So, assumed that winter means snow everywhere. At least it looks like it on the tileset. But the tile description still says: dirt, grass, etc.
Function `game::is_in_sunlight` considers only two type of weather as `sunlight`: `WEATHER_CLEAR` and `WEATHER_SUNNY`. Looks correct to me.

And `snow` weather means that it is snowing right now (not sunny or clear). So it should not lead to sun glare, right?

Refactored `weather_effect::glare` to avoid mistakes when adding other glare-related effects, like checking for specific mutations.
Now `glare` is called for both clear and sunny weather, but now for snow. To keep original behavior mostly intact, clear weather only add glare effect during winter (assuming snow everywhere).

Removed `weather_effect::snow_glare()`, can't find any usage of it.

Tested with various combinations:
- day / night
- inside / outside
- clear / sunny / show weather
- spring / winter
- with / without `CEPH_VISION` mutation

#### Describe alternatives you've considered
The same, but keeping the original behavior for `snow` weather (causing glare). Still seems wrong to me.

#### Additional context
Using only logic here, no actual IRL reference. 
Probably incorrect solution to remove glare from snow weather?

ipcyborg commented:
Jenkins Rebuild

BevapDin commented on src/weather.cpp:
```diff
@@ -61,44 +61,37 @@ static bool is_player_outside()
+    if( season == WINTER ) {
+        //Winter snow glare: for both clear & sunny weather
+        effect = &effect_snow_glare;
+        dur = ( !g->u.has_effect( *effect ) ) ? 2_turns : 1_turns;
```
```suggestion
        dur = g->u.has_effect( *effect ) ? 1_turns : 2_turns;
```
Same below. It's just easier to read and avoids double negation.

BevapDin commented on src/weather.cpp:
```diff
@@ -61,44 +61,37 @@ static bool is_player_outside()
 /**
  * Glare.
  * Causes glare effect to player's eyes if they are not wearing applicable eye protection.
+ * @param intensity Level of sun brighthess (1 = clear, 2 = sunny)
```
Seems like this should be an `enum`, as hinted by the explanation in the comment: there are only two possible values and you have given them names in the comment. Why not give those two values names in the code itself?

ipcyborg replied:
Fixed, thanks

davidpwbrown commented:
Snowglare was removed in #28371 
Or, at least I thought it was.

ipcyborg commented:
> Snowglare was removed in #28371

That commit was about removing `glare` function from `weather_effect::snow`. 

>Snowglare logic was inconistent.

Totally agree with it. So I just did the same, removed `glare` from `weather_effect::snow`. But also added glare for sunny & clear weather in winter. 
It is not perfect, but I think it is more logical that the current state.

Alternative: just remove snowglare completely.

davidpwbrown commented:
> > Snowglare was removed in #28371
> 
> That commit was about removing `glare` function from `weather_effect::snow`.
> 
> > Snowglare logic was inconistent.
> 
> Totally agree with it. So I just did the same, removed `glare` from `weather_effect::snow`. But also added glare for sunny & clear weather in winter.
> It is not perfect, but I think it is more logical that the current state.
> 
> Alternative: just remove snowglare completely.

Thats weird then, unless im missing something. 

`void weather_effect::snow()
{
    wet_player( 10 );
    glare( true );
}`  glare(true) should have been removed ages ago, and yet it is somehow back there!

ipcyborg commented:
>the glare( true ) should have been removed ages ago, and yet it is somehow back there!

Looks like that PR #28371 was never committed, is was just closed:

>Merged to master, then merged and reverted to development.
>kevingranade closed this on Feb 25

However, the same change is also included in this PR.
